library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java

```diff
@@ -378,30 +378,50 @@
     private final ContentResolver contentResolver;
     private final DownloadsUriProvider downloadsUriProvider;
     private final SystemFacade systemFacade;
+    private final BatchRepository batchRepository;
 
     private Uri baseUri;
 
-    public DownloadManager(Context context, ContentResolver resolver) {
-        this(context, resolver, DownloadsUriProvider.getInstance(), new RealSystemFacade(context), false);
+    public static DownloadManager newInstance(Context context, ContentResolver contentResolver) {
```

Comment:
yeah but you are then shrinking the library capabilities in order to simplify what we have to think about. Descoping isn't possible because of the package structure of the libarary, and that's the point people can use the builder it's the number 1 simple route, but the public downloadmanager constructor should still be available and be the entry point into this package.